graphql/redact_secrets_plugin.go

```diff
@@ -0,0 +1,111 @@
+package graphql
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"sort"
+
+	"github.com/99designs/gqlgen/codegen/config"
+)
+
+// GenerateSecretFields generates a file that contains a list of fields that should be redacted in logs.
+func GenerateSecretFields(cfg *config.Config) error {
+	redactedFields := make(map[string]bool)
+
+	for _, schemaType := range cfg.Schema.Types {
+		for _, field := range schemaType.Fields {
+			if field.Directives.ForName("redactSecrets") != nil {
+				redactedFields[field.Name] = true
+			}
+		}
+	}
+
+	var fields []string
+	for field := range redactedFields {
+		fields = append(fields, field)
+	}
+
+	// Sort the fields to ensure consistent output.
+	sort.Strings(fields)
+
+	return generateRedactedFieldsFile(fields)
+}
+
+// generateRedactedFieldsFile generates a file that contains a list of fields that should be redacted in logs. It is used to generate the redacted_fields_gen.go file.
+// The file contains a list of fields that should be redacted in logs, a function to check if a field should be redacted, and a function to redact fields in a map.
+func generateRedactedFieldsFile(fields []string) error {
+	file, err := os.Create("graphql/redacted_fields_gen.go")
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	_, err = file.WriteString(`// Code generated by graphql/redact_secrets_plugin.go DO NOT EDIT.
+package graphql
+
+var redactedFields = []string{
+`)
+	if err != nil {
+		return err
+	}
+
+	for _, field := range fields {
+		_, err = file.WriteString(fmt.Sprintf("\t\"%s\",\n", field))
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err = file.WriteString("}\n")
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+func isFieldRedacted(fieldName string, fieldsToRedact []string) bool {
+	for _, field := range fieldsToRedact {
+		if field == fieldName {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
would there be a performance boost by using a map[string]bool for `fieldsToRedact`?

Reply:
There would especially as we continue to redact new fields